File: shared/src/main/java/ru/privatenull/pnauth/kernel/event/DecisionEvent.java
package ru.privatenull.pnauth.kernel.event;
import java.util.List;
public interface DecisionEvent extends ExtensionEvent {
    boolean cancelled();
    void setCancelled(boolean cancelled);
    default void cancel() { setCancelled(true); }
    default void allow() { setCancelled(false); }
    void setDecision(boolean cancelled, String reason);
    EventDecision effectiveDecision();
    List<EventDecision> decisionHistory();
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/kernel/event/DecisionEvent.kt
package ru.privatenull.pnauth.kernel.event

interface DecisionEvent : ExtensionEvent {
    fun cancelled(): Boolean
    fun setCancelled(cancelled: Boolean)
    fun cancel() { setCancelled(true) }
    fun allow() { setCancelled(false) }
    fun setDecision(cancelled: Boolean, reason: String?)
    fun effectiveDecision(): EventDecision
    fun decisionHistory(): List<EventDecision>
}
